packages/pricing/src/utils/validate-price-list-dates.ts

```diff
@@ -0,0 +1,26 @@
+import { isDate, MedusaError } from "@medusajs/utils"
+
+export const validatePriceListDates = (priceListData: {
+  starts_at?: Date | string | null
+  ends_at?: Date | string | null
+}) => {
+  if (
+    typeof priceListData.starts_at === "string" &&
+    !isDate(priceListData.starts_at)
+  ) {
+    throw new MedusaError(
+      MedusaError.Types.INVALID_DATA,
+      `Cannot set price list starts at with with invalid date string: ${priceListData.starts_at}`
+    )
+  }
+
+  if (
+    typeof priceListData.ends_at === "string" &&
+    !isDate(priceListData.ends_at)
+  ) {
+    throw new MedusaError(
+      MedusaError.Types.INVALID_DATA,
+      `Cannot set price list ends at with with invalid date string: ${priceListData.ends_at}`
+    )
+  }
+}
```

Comment:
I would argue we use utils that's about validating a date to achieve the goal of validating a price-list configuration in this util. 

I think the only way to make this more generic without losing the semantics would creating a `isValidDateIfString` util to combine the if conditions in one function call, otherwise we start passing along exception messages as well (or catch and rethrow to match the control flow). 

I've moved this to the utils package just in case we want to use it in the core as well.

Reply:
I would counter argue that isValidOrThrow would do what you say. But for me, isDateValid would return true or false and allow you to integrate in your control flow. It is not necessarily suppose to throw. Here the util is basically just checking that an input is indeed a date. I would then in your flow do something like if (!isDateValid) { throw }

Thats my thinking. It is not a strong point though, but i can see this utils being used in a lot of modules and else where to match our way to handle date